src/normalizer.ts

```
@@ -222,6 +223,32 @@ rules.set('Transform const to singleton enum', schema => {
   }
 })
 
+rules.set('Transform nullable to null type', schema => {
+  if (schema.nullable !== true) {
+    return
+  }
+
+  delete schema.nullable
+
+  if (schema.const !== undefined) {
+    if (schema.const !== null) {
+      schema.enum = [schema.const, null]
+      delete schema.const
+    }
+  } else if (schema.enum) {
+    if (!schema.enum.includes(null)) {
+      schema.enum.push(null)
+      log('yellow', 'normalizer', 'enum should include null when schema is nullable', schema)
```

@mdmower-csnw: 
1. Should there be a warning for the `const` case, too? [AJV](https://ajv.js.org/json-schema.html#nullable) docs say with `nullable: true`:
> const would fail, unless it is "const": null

2. I wouldn't be opposed to throwing instead of the warnings, because it's critical that the schema is updated or the type will be at odds with how AJV validates.

3. What if `const` and `enum` are both defined? Is that case caught earlier by validation?

---

The `const` case is mostly moot because normalizer _Transform const to singleton enum_ runs just before _Transform nullable to null type_:

```ts
rules.set('Transform const to singleton enum', schema => {
  if (schema.const !== undefined) {
    schema.enum = [schema.const]
    delete schema.const
  }
})
```

I included `const` handling just in case normalizers are reordered in the future.

> What if const and enum are both defined? Is that case caught earlier by validation?

`schema.enum` actually gets wiped out in this case due to the above normalizer. This is a great example of how this tool is not a validator. There's an expectation that the user has performed validation prior to using this tool.

> Should there be a warning for the const case, too? [AJV](https://ajv.js.org/json-schema.html#nullable) docs say with `nullable: true`.

That's fine. Can do.

> I wouldn't be opposed to throwing instead of the warnings, because it's critical that the schema is updated or the type will be at odds with how AJV validates.

Like mentioned before, this is not a validator. It's a tool that seldom throws.

---

Opted to reorder transforms so that _Transform nullable to null type_ runs before _Transform const to singleton enum_ in [a5db6da](https://github.com/CSNW/json-schema-to-typescript/pull/1/commits/a5db6da52b5992917f08f911994d8ed2df58f1b5). Also created a `warning()` function that prints to the console even when `VERBOSE` is not used.